src/Microsoft.AspNet.Mvc.Core/Filters/FormatFilterAttribute.cs

```diff
@@ -0,0 +1,115 @@
+using Microsoft.Net.Http.Headers;
+using System.Diagnostics;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// This will look at the format parameter if present in the route data or query data and 
+    /// sets the content type in ObjectResult corresponding to the format value.
+    /// </summary>
+    public class FormatFilterAttribute : Attribute, IFormatFilter, IResourceFilter, IResultFilter
+    {
+        /// <summary>
+        /// As a resourceFilter, this filter looks at the request and rejects it
+        /// before going ahead if
+        /// 1. The Format in the request doesnt match any format in the map.
+        /// 2. If there is a conflicting producesFilter.
+        /// </summary>
+        /// <param name="context"></param>
+        public void OnResourceExecuting([NotNull] ResourceExecutingContext context)
+        {
+            var format = GetFormat(context);
+
+            if (format != null)
+            {
+                var contentType = GetContentType(format, context);
+                if (contentType == null)
+                {
+                    // no contentType exists for the format, return 404
+                    context.Result = new HttpNotFoundResult();
+                }
+                else
+                {
+                    var produces = context.Filters.LastOrDefault(f => f is ProducesAttribute) as ProducesAttribute;
+                    if (produces != null && !produces.ContentTypes.Any(c => c.IsSubsetOf(contentType)))
+                    {
+                        context.Result = new HttpNotFoundResult();
+                    }
+                }
+            }
+        }
+
+        public void OnResourceExecuted([NotNull] ResourceExecutedContext context)
+        {
+
+        }
+
+        public void OnResultExecuting([NotNull] ResultExecutingContext context)
+        {
+            var format = GetFormat(context);
+            if (format != null)
+            {
+                var contentType = GetContentType(format, context);
+                Debug.Assert(contentType != null);
+
+                var objectResult = context.Result as ObjectResult;
+                if (objectResult != null)
+                {
+                    objectResult.ContentTypes.Clear();
+                    objectResult.ContentTypes.Add(contentType);
+                }
+            }
+        }
+
+        public void OnResultExecuted([NotNull] ResultExecutedContext context)
+        {
+
+        }
+
+        public MediaTypeHeaderValue GetContentTypeForCurrentRequest(FilterContext context)
+        {
+            var format = GetFormat(context);
+            if (format != null)
+            {
+                return GetContentType(format, context);
+            }
+
+            return null;
+        }
+
+        private object GetFormat(FilterContext context)
+        {
+            object format = null;
+
+            if (!context.RouteData.Values.TryGetValue("format", out format))
```

Comment:
suggest to add a class level constant somewhere for format so that we need to change it at one place.